numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -1583,7 +1583,7 @@ _array_fromobject(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *kws)
 
     if (PyTuple_GET_SIZE(args) > 2) {
         PyErr_SetString(PyExc_ValueError,
-                        "only 2 non-keyword arguments accepted");
+                        "array() takes from 1 to 2 positional arguments but %zd were given", PyTuple_GET_SIZE(args)));
```

Comment:
```suggestion
                     "array() takes from 1 to 2 positional arguments but %zd were given", PyTuple_GET_SIZE(args)));
```
to match the indentation change above